update links and formatting, project landing pages

no-stack-dub-sack opened the pull request:
<!-- freeCodeCamp Pull Request Template -->

<!-- IMPORTANT Please review https://github.com/freeCodeCamp/freeCodeCamp/blob/staging/CONTRIBUTING.md for detailed contributing guidelines -->
<!-- Help with PRs can be found at https://gitter.im/FreeCodeCamp/Contributors -->
<!-- Make sure that your PR is not a duplicate -->

#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply. -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change
<!-- What type of change does your code introduce? After creating the PR, tick the checkboxes that apply. -->
- [x] curriculum update
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask in the Help Contributors room linked above. We're here to help! -->
- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #12661

#### Description
<!-- Describe your changes in detail -->
@QuincyLarson  this PR addresses #12661:
- updates all project landing pages with correct test suite CDN link
- updates all project landing pages with correct test suite code pen links
- added Tree Map D3 project 
- added dataset links for all D3 projects

There's still a bit to be done with the landing pages, so we will need to open a separate issue for that and/or condense already existing issues

no-stack-dub-sack committed:
update links and formatting, project landing pages

raisedadead commented on seed/challenges/03-front-end-libraries/front-end-libraries-projects.json:
```diff
@@ -96,8 +96,10 @@
         "Here's a <a href='http://codepen.io/freeCodeCamp/full/wgGVVX' target='_blank'>working example</a>. Try not to look at its code.",
-        "You can build your project by forking <a href='http://codepen.io/FCC_test_suites/pen/JbgXab' target='_blank'>this CodePen pen</a>. Or you can use this CDN link to run the tests in any environment you like: <code>https://gitcdn.link/repo/no-stack-dub-sack/testable-projects-fcc/master/build/bundle.js</code>",
-        "Once you're done, submit below the URL to your working project with all its tests passing.",
+        "<strong>Note On Calculator Logic:</strong> It should be noted that there are two main schools of thought on calculator input logic: <dfn>immediate execution logic</dfn> and <dfn>formula logic</dfn>. Our example utilizes formula logic and observes order of operation precedence, immediate execution does not. Either is acceptable, but please note that depending on which you choose, your cal ...
```
Break this into multiple lines:

![image](https://cloud.githubusercontent.com/assets/1884376/22309986/9995faaa-e373-11e6-94bb-ba2d376d3db4.png)

raisedadead commented on seed/challenges/04-data-visualization/data-visualization-projects.json:
```diff
@@ -10,8 +10,9 @@
         "Here's a <a href='http://codepen.io/freeCodeCamp/full/GrZVaM' target='_blank'>working example</a>. Try not to look at its code.",
-        "You can build your project by forking <a href='http://codepen.io/FCC_test_suites/pen/JbgXab' target='_blank'>this CodePen pen</a>. Or you can use this CDN link to run the tests in any environment you like: <code>https://gitcdn.link/repo/no-stack-dub-sack/testable-projects-fcc/master/build/bundle.js</code>",
-        "Once you're done, submit below the URL to your working project with all its tests passing.",
+        "Here is the dataset you will need to complete this project:<br><ul><li><code>https://raw.githubusercontent.com/FreeCodeCamp/ProjectReferenceData/master/GDP-data.json1</code></li></ul>",
```
Inconsistent bullet points

![image](https://cloud.githubusercontent.com/assets/1884376/22310105/076fbc5a-e374-11e6-8577-d2c11d05fe05.png)

raisedadead commented:
Do we really need bullet points, I am not a huge fan but this could just simply follow the same format as that of the previous JS and HTML challenges?

We already have less real estate, with the large font size. What do you think?

no-stack-dub-sack commented:
@raisedadead I'm ok with going no bullet points, I only did it there for consistency with the D3 projects that have multiple datasets which I felt needed them: 
![image](https://cloud.githubusercontent.com/assets/18563015/22314461/88353620-e32f-11e6-98d0-e08aeb5db0e9.png)

But I guess I could just leave them there where they make sense, and get rid of them everywhere else. Sound good? Or do you think I should just ditch them there too? I thought, for that one and the other with multiple, that it was important to let the user know what the particular dataset was for

raisedadead commented:
Yeah, I think the label on top of the resources is important. And I also agree that you should leave the bullet points where they make sense as in the example above.

no-stack-dub-sack commented:
@raisedadead ok, so I'll change it to be in only those challenges that have multiple datasets, and leave the rest without bullet points. I still think it looks a little funny, be was have bigger things to worry about for the time being so this will do for now.

raisedadead commented:
I think this needs a styling overhaul. but yeah if you could address the first comment about breaking the instructions to at least multiline, we can ignore the other aesthetics for now.

no-stack-dub-sack commented:
@raisedadead about to make a PR and I think this does it for now - added the multiline for the example like you said, and removed the bullets where they were not needed (from every challenge except D3 tree map and D3 choropleth where multiple datasets are needed)

Also, I definitely agree with you that these sections need an overall style overhaul 😄

no-stack-dub-sack commented:
@raisedadead WAIT! Don't merge this yet - looks like #12720 affects this entire section, so once that PR is merged, I will rebase off of that before this gets merged. Does that make sense or should the merge still go ok if this is merged first?

raisedadead commented:
Blocked for merge.

raisedadead commented:
@no-stack-dub-sack I have resolved a merge conflict with a PR that came in with another PR https://github.com/freeCodeCamp/freeCodeCamp/pull/12909

Can you please do a sanity check and let me know, for a merge?

Also we need not wait for #12720, it already has conflicts and should be resolved separately.

no-stack-dub-sack commented:
@raisedadead Yup, if all you changed is adding the closing `</a>` tags to resolve that conflict than LGTM!
